docs/contributing/advanced_development_topics.rst

```diff
@@ -0,0 +1,156 @@
+.. -*- mode: rst -*-
+
+Advanced Development Topics
+===========================
+
+In this section we discuss more advanced contributing guidelines such as code conventions,the release life cycle or branch management. This section is intended for maintainers and core contributors of the Yellowbrick project. If you would like to be a maintainer please contact one of the current maintainers of the project.
+
+Visualizer Review Checklist
+---------------------------
+
+As the visualizer API has matured over time, we've realized that there are a number of routine items that must be in place to consider a visualizer truly complete and ready for prime time. This list is also extremely helpful for reviewing code submissions to ensure that visualizers are consistently implemented, tested, and documented. Though we do not expect these items to be checked off on every PR, the below list includes some guidance about what to look for when reviewing or writing a new Visualizer.
+
+Code Conventions
```

Comment:
These are a great collection of learnings - almost worthy of a pydata talk =).  

One thing that I would add is trying to keep the API organized where it makes sense, ie matplotlib in `draw`, scikit-learn model fitting  in `fit()`, etc. I think this captured in other docs but a little repetition here might be helpful for newer reviewers

Reply:
Added it to the list of conventions!